dvc_azure/spec.py

```diff
@@ -4,10 +4,10 @@
 class AzureBlobFileSystem(  # pylint: disable=abstract-method
     _AzureBlobFileSystem
 ):
-    async def _put_file(self, *args, **kwargs) -> None:
+    def put_file(self, *args, **kwargs) -> None:
         kwargs["overwrite"] = True
-        return await super()._put_file(*args, **kwargs)
+        return super().put_file(*args, **kwargs)
```

Comment:
This breaks the case where we use the async coroutines directly in batched file operations. We need to be setting `overwrite` in both the sync `put_file` and async `_put_file` versions of the method.